### ISSUE TITLE
Document Twisted Words examples

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5589]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)| @ashfall reported|
|-|-|
|Trac ID|trac#5589|
|Type|task|
|Created|2012-04-01 20:47:06Z|
|Branch|https://github.com/twisted/twisted/tree/document-words-examples-5589|

Documentation on how to run the Twisted Words examples at http://twistedmatrix.com/documents/current/words/examples/index.html and description for each of the scripts should be added.

Attachments:

* [Ticket5888TwistedWordsExamplesAddDocumentationSecondTry.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/603/60396c36c408b59884b04c53e9b6ee44b8f9d7a1/578403cb39f5f0db29ce2cfd0575e3504f7c201c.patch) (24069 bytes) - added by ashfall on 2012-11-17 18:15:54Z - Patch by anitgen (from #5888)
* [5589.doc](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/603/60396c36c408b59884b04c53e9b6ee44b8f9d7a1/cfd56eec97d2eed05ab8b5eebbefc7a412daadb6.doc) (96 bytes) - added by anitgen on 2012-11-26 15:45:54Z - Newsfile
* [5589.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/603/60396c36c408b59884b04c53e9b6ee44b8f9d7a1/622a39ab32f4680c527114d05e7d7e50c4a01c91.patch) (3384 bytes) - added by thijs on 2012-12-07 20:52:11Z - actual patch from #5888

<details><summary>Searchable metadata</summary>

```
trac-id__5589 5589
type__task task
reporter__ashfall ashfall
priority__high high
milestone__ 
branch__branches_document_words_examples_5589 branches/document-words-examples-5589
branch_author__ashfall__jml ashfall, jml
status__closed closed
resolution__fixed fixed
component__words words
keywords__documentation_easy documentation easy
time__1333313226000000 1333313226000000
changetime__1355821907000000 1355821907000000
version__None None
owner__jml jml
cc__jesstess cc__ralphm cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)|@ashfall removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_3"></a>|@ashfall commented|
|-|-|

Patch attached, I guess it's meant for review.

Needs a news file, though.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_4"></a>|@jml commented|
|-|-|

(In [36584]) Branching to 'document-words-examples-5589'

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml commented|
|-|-|

(In [36587]) Consistent indentation of literal blocks in docstrings (refs [#5589](https://github.com/twisted/twisted/issues/5589))

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_6"></a>|@jml commented|
|-|-|

I've made a branch for this and made the following changes in lieu of a review:

 * Consistent indentation on code blocks
 * Drop 'twisted' from 'this simple twisted example' in `msn_example.py`, since it's implied by the context
 * Capitalize "twisted" and "jabber" in `ircLogBot.py`

I'll double check with another dev & land this.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set owner to @jml|
|-|-|

The patch adds a reference to `twisted.oscar` in `oscardemo.py`, but this doesn't exist and should be `twisted.words.protocols.oscar` instead. Other than that it looks good to me, please merge after that.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_8"></a>|@jml commented|
|-|-|

(In [36595]) twisted.oscar -> twisted.words.protocols.oscar

Thanks therve!  (refs [#5589](https://github.com/twisted/twisted/issues/5589))

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_9"></a>|@jml set status to `closed`|
|-|-|

(In [36596]) Merge document-words-examples-5589: Explain how to use words examples

Author: anitgen, jml
Reviewers: ashfall, thijs, therve
Fixes: [#5589](https://github.com/twisted/twisted/issues/5589)
